### DISCUSSION TITLE
Update minimal pandas version up to 2.0

### DISCUSSION THREAD

**Discussion by d-a-bunin** [etna-team/etna#421] [Improvements]
### Description

Currently (2.7.1) the library supports pandas versions: `^1.1`. It means that we don't support pandas-2.0.

If we remove this limitation, some dependencies are still preventing us from using pandas `>2`, e.g. `pytorch_forecasting`.

There are some places in a code when we made fixes only for old versions of pandas:
- https://github.com/etna-team/etna/blob/4a8bbb553a57304fb50f21f49b52fa2e780bd66c/etna/datasets/tsdataset.py#L1393
- https://github.com/etna-team/etna/blob/4a8bbb553a57304fb50f21f49b52fa2e780bd66c/etna/datasets/tsdataset.py#L1397
- https://github.com/etna-team/etna/blob/4a8bbb553a57304fb50f21f49b52fa2e780bd66c/etna/datasets/tsdataset.py#L1469 
- https://github.com/etna-team/etna/blob/4a8bbb553a57304fb50f21f49b52fa2e780bd66c/etna/datasets/tsdataset.py#L1522
- https://github.com/etna-team/etna/blob/4a8bbb553a57304fb50f21f49b52fa2e780bd66c/etna/reconciliation/base.py#L97
- https://github.com/etna-team/etna/blob/4a8bbb553a57304fb50f21f49b52fa2e780bd66c/etna/reconciliation/base.py#L101 
- https://github.com/etna-team/etna/blob/4a8bbb553a57304fb50f21f49b52fa2e780bd66c/etna/transforms/encoders/mean_segment_encoder.py#L78
- https://github.com/etna-team/etna/blob/4a8bbb553a57304fb50f21f49b52fa2e780bd66c/etna/datasets/tsdataset.py#L1002

### Thoughts

I'm not really sure if this is a good idea, because this is a very big upgrade of a minimal version.

### Related

- #399
- TODO: link discussion about `pytorch_forecasting` after creation

**Comment by egoriyaa** [comment]
Solved in #599
